hdmi/tinyaudio_hw.c

```diff
@@ -228,22 +232,24 @@ static int start_output_stream(struct stream_out *out)
     struct audio_device *adev = out->dev;
     struct pcm_params *params;
     int device = 0;
-
     ALOGV("%s enter",__func__);
 
     if ((adev->card < 0) || (adev->device < 0)){
         char value[PROPERTY_VALUE_MAX];
 
         /*this will be updated once the hot plug intent
           sends these information.*/
+        /* hard coded assignment of card can be removed with
+           parse_hdmi_device_number implementation */
         adev->card = DEFAULT_CARD;
         property_get("ro.vendor.hdmi.audio", value, "0");
         if (!strcmp(value,"ehl")) {
             adev->device = DEFAULT_DEVICE_EHL;
```

Comment:
Should we move these hardcodings below 'parse_hdmi_device_number'(only when it returns -1).